integration-tests/src/test/java/org/apache/druid/tests/indexer/AbstractITRealtimeIndexTaskTest.java

```diff
@@ -88,25 +89,30 @@ void doTest()
       // the task will run for 3 minutes and then shutdown itself
       String task = setShutOffTime(
           getResourceAsString(getTaskResource()),
-          DateTimes.utc(System.currentTimeMillis() + TimeUnit.MINUTES.toMillis(3))
+          DateTimes.utc(System.currentTimeMillis() + TimeUnit.MINUTES.toMillis(10))
       );
       task = StringUtils.replace(task, "%%DATASOURCE%%", fullDatasourceName);
 
       LOG.info("indexerSpec: [%s]\n", task);
       String taskID = indexer.submitTask(task);
 
-
       // sleep for a while to let peons finish starting up
-      TimeUnit.SECONDS.sleep(5);
+      TimeUnit.SECONDS.sleep(60);
```

Comment:
why 60? can we use `ITRetryUtil.retryUntil...` instead of making every test wait 1 minute?

Reply:
We don't have any way to check whether the task is ready to consume data or not for now. 5 seconds was too short and I think 1 min of waiting isn't that harm in a sense that our tests usually runs for 30 mins.

Reply:
In my experience, once a sleep is added, it's pretty hard to remove a sleep once it's added to the tests. At the very least, we should add a comment here saying why we need to sleep and how we could fix this in the future if the readiness endpoint existed.

Try till success, means we'd wait at most 1 minute. Right now every time our tests will run for 31 minutes instead of 30. Happy to discuss this offline - maybe it's not such a big deal if we add an extra minute to the tests

Reply:
I agree with you on adding a sleep. But `RealtimeIndexTask` and `AppenderatorRealtimeIndexTask` are sort of deprecated (in a sense that Tranquility is deprecated which is the main user of those tasks) and I don't think it's worth to spend much time for them. 

Regarding the test run time, I think the most 2 common cases when the test run time matters are the CI and manual running of tests. For CI, LGTM and Teamcity are running more than an hour and Travis is not a bottleneck. For manual running of tests, I usually do something else while tests are running and adding 1 more min doesn't bother me much. What do you think?

BTW, anything happened only offline is regarded as that hasn't happened in [Apache way](https://www.apache.org/theapacheway/). 

> Open Communications: as a virtual organization, the ASF requires all communications related to code and decision-making to be publicly accessible to ensure asynchronous collaboration, as necessitated by a globally-distributed community. 

We can discuss offline, but that discussion and the conclusion must be filed somewhere public.

Reply:
Discussed offline - these tests are for parts of the codebase that is currently deprecated. They will likely be removed in the near future. Right now the bottleneck for PR checks is LGTM (90 min) So even though this adds 2 min extra - there is no visible difference to the time it takes to merge a PR.